test_prebuild/test_mpi_test/data.F90

```diff
@@ -0,0 +1,32 @@
+module data
+
+!! \section arg_table_data Argument Table
+!! \htmlinclude data.html
+!!
+    use ccpp_types,    only: ccpp_t
+
+#if defined MPI_F08
+#define ccpp_mpi_comm_type type(mpi_comm)
+#else
+#define ccpp_mpi_comm_type integer
+#endif
+
+#if defined MPI_F08
+    use mpi_f08
+#elif defined MPI_F90
+    use mpi
+#endif
+
+    implicit none
+
+    private
+
+    public ncols
+    public ccpp_data
+    public ccpp_mpi_comm
+
+    integer, parameter :: ncols = 1
+    type(ccpp_t), target :: ccpp_data
+    ccpp_mpi_comm_type :: ccpp_mpi_comm
```

Comment:
Add type(

Reply:
Exactly NOT - this doesn't work if it is an integer. That's the whole point behind the preprocessor logic and making ccpp_mpi_comm_type look like an intrinisic variable type.

Reply:
Duh. My bad. The syntax caught me off guard.

Reply:
It is a bit weird, yes, and it will require good documentation for CCPP physics developers and users.